app/components/sidebars/main/channels_list/list/list.js

```diff
@@ -164,14 +176,36 @@ export default class List extends PureComponent {
     };
 
     buildSections = (props) => {
+        if (props.showLegacySidebar) {
+            const {
+                orderedChannelIds,
+            } = props;
+
+            return orderedChannelIds.map((s) => {
+                return {
+                    ...this.getSectionConfigByType(props, s.type),
+                    data: s.items,
+                };
+            });
+        }
+
         const {
+            channelsByCategory,
             orderedChannelIds,
         } = props;
 
-        return orderedChannelIds.map((s) => {
+        const unreads = orderedChannelIds.find((el) => el.type === 'unreads');
+
+        channelsByCategory.unshift({
+            channel_ids: unreads.items,
+            name: unreads.name,
+            type: unreads.type,
+        });
```

Comment:
A quick workaround to get unreads into category view - I think there was a getUnreadChannels method in the mm-redux I can replace this with